tools/RAiDER/models/ecmwf.py

```diff
@@ -52,24 +53,28 @@ def setLevelType(self, levelType):
         else:
             self.__pressure_levels__()
 
+
     def __pressure_levels__(self):
         self._zlevels = np.flipud(LEVELS_25_HEIGHTS)
         self._levels = len(self._zlevels)
 
+
     def __model_levels__(self):
         self._levels = 137
         self._zlevels = np.flipud(LEVELS_137_HEIGHTS)
         self._a = A_137_HRES
         self._b = B_137_HRES
 
-    def load_weather(self, *args, **kwargs):
+
+    def load_weather(self, f=None, *args, **kwargs):
         '''
         Consistent class method to be implemented across all weather model types.
         As a result of calling this method, all of the variables (x, y, z, p, q,
         t, wet_refractivity, hydrostatic refractivity, e) should be fully
         populated.
         '''
-        self._load_model_level(*self.files)
+        f = self.files[0] if f is None else f
```

Comment:
i've updated this throughout the weather model files (and adjusted all to use `f` instead of `filename`) to be able to accept a file input